browser/android/brave_cosmetic_resources_tab_helper.cc

```diff
@@ -0,0 +1,123 @@
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "brave/browser/android/brave_cosmetic_resources_tab_helper.h"
+
+#include <memory>
+#include <string>
+#include <utility>
+
+#include "brave/browser/brave_browser_process_impl.h"
+#include "brave/components/brave_shields/browser/ad_block_custom_filters_service.h"
+#include "brave/components/brave_shields/browser/ad_block_regional_service_manager.h"
+#include "brave/components/brave_shields/browser/ad_block_service.h"
+#include "brave/components/brave_shields/browser/ad_block_service_helper.h"
+#include "brave/components/brave_shields/browser/brave_shields_util.h"
+#include "chrome/browser/content_settings/host_content_settings_map_factory.h"
+#include "chrome/browser/profiles/profile.h"
+#include "content/browser/web_contents/web_contents_impl.h"
+#include "content/public/browser/navigation_handle.h"
+#include "content/public/renderer/render_frame.h"
+
+
+namespace {
+bool ShouldDoCosmeticFiltering(content::WebContents* contents,
+    const GURL& url) {
+  Profile* profile = static_cast<Profile*>(contents->GetBrowserContext());
+  auto* map = HostContentSettingsMapFactory::GetForProfile(profile);
+
+  return ::brave_shields::ShouldDoCosmeticFiltering(map, url);
+}
+
+std::unique_ptr<base::ListValue> GetUrlCosmeticResourcesOnTaskRunner(
+    const std::string& url) {
+  auto result_list = std::make_unique<base::ListValue>();
+
+  base::Optional<base::Value> resources = g_brave_browser_process->
+      ad_block_service()->UrlCosmeticResources(url);
+
+  if (!resources || !resources->is_dict()) {
+    return result_list;
+  }
+
+  base::Optional<base::Value> regional_resources = g_brave_browser_process->
+      ad_block_regional_service_manager()->UrlCosmeticResources(url);
+
+  if (regional_resources && regional_resources->is_dict()) {
+    ::brave_shields::MergeResourcesInto(
+        std::move(*regional_resources), &*resources, /*force_hide=*/false);
+  }
+
+  base::Optional<base::Value> custom_resources = g_brave_browser_process->
+      ad_block_custom_filters_service()->UrlCosmeticResources(url);
+
+  if (custom_resources && custom_resources->is_dict()) {
+    ::brave_shields::MergeResourcesInto(
+        std::move(*custom_resources), &*resources, /*force_hide=*/true);
+  }
+
+  result_list->Append(std::move(*resources));
+
+  return result_list;
+}
+
+void GetUrlCosmeticResourcesOnUI(content::RenderFrameHost* render_frame_host,
+  std::unique_ptr<base::ListValue> resources) {
+  if (!resources) {
+    return;
+  }
+  for (auto i = resources->GetList().begin();
+      i < resources->GetList().end(); i++) {
+    base::DictionaryValue* resources_dict;
+    if (!i->GetAsDictionary(&resources_dict)) {
+      continue;
+    }
+    std::string to_inject;
+    resources_dict->GetString("injected_script", &to_inject);
+    if (to_inject.length() > 1) {
+      content::RenderFrameHost::AllowInjectingJavaScript();
```

Comment:
I'm wondering we can call this api freely because upstream doesn't use like us.
I know we've used this api so far. Just curious this is safe. @bridiver ?
```
  // Globally allows for injecting JavaScript into the main world. This feature
  // is present only to support Android WebView, WebLayer, Fuchsia web.Contexts,
  // and CastOS content shell. It must not be used in other configurations.
```

Reply:
@SergeyZhukovsky since these scripts modify the DOM (which is shared across isolated worlds), can we use `ExecuteJavaScriptInIsolatedWorld` instead?

Reply:
Remade it to use `ExecuteJavaScriptInIsolatedWorld`. @simonhong @bridiver could you please check that?